src/Devices/OWSReadReceiptsProcessor.m

```diff
@@ -90,6 +90,10 @@ - (void)process
             // Mark all unread messages in this thread that are older than message specified in the read
             // receipt.
             NSMutableArray<id<OWSReadTracking>> *interactionsToMarkAsRead = [NSMutableArray new];
+
+            // Always mark the message specified by the read receipt as read.
+            [interactionsToMarkAsRead addObject:message];
```

Comment:
Yes, this should be redundant but it felt safer to be a little redundant. There's just enough complexity here that I could see future refactoring elsewhere breaking this logic.